lib/promoted/ruby/client/request_builder.rb

```diff
@@ -48,10 +48,16 @@ def delivery_request_params(should_compact: true)
           params = {
```

Comment:
side-comment - In promoted-ts-client, I think I clone the Request input and then modify it with these.  It reduces the chance that we forget to pass through fields here.

Reply:
Got it; At some point I will overhaul the compaction code to separate out metrics and delivery and at that point this is an easy change.